### PR TITLE
Spring 5

### PR DESCRIPTION
**Pull request by Fedich09**

### CODE REVIEW

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        return movieSessionService.findAvailableSessions(id,
+                LocalDate.parse(showTime, formatter)).stream()
+                .map(sessionMapper::toDto)
+                .collect(Collectors.toList());
+    }
+
+    @PutMapping
+    public void update(@RequestBody MovieSessionRequestDto movieSessionRequestDto,
+                       @RequestParam Long id) {
```
Comment (semyonich) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]: id should be PathVariable, not param
HTTP PUT http://api.example.com/device-management/managed-devices/{id}  //Update device for given Id
HTTP DELETE http://api.example.com/device-management/managed-devices/{id}  //Delete device for given Id

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        MovieSession movieSession = sessionMapper.toEntity(movieSessionRequestDto);
+        movieSession.setId(id);
+        movieSessionService.update(movieSession);
+    }
+
+    @DeleteMapping
+    public void delete(@RequestParam Long id) {
```
Comment (semyonich) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]: same

**Review by semyonich — CHANGES_REQUESTED** [review]

**Review by semyonich — APPROVED** [review]

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    private final CinemaHallMapperImpl cinemaHallMapperImpl;
+    private final CinemaHallService cinemaHallService;
+
+    public CinemaHallController(CinemaHallMapperImpl cinemaHallMapperImpl,
+                                CinemaHallService cinemaHallService) {
+        this.cinemaHallMapperImpl = cinemaHallMapperImpl;
+        this.cinemaHallService = cinemaHallService;
+    }
+
+    @PostMapping()
```
Comment (boroda4436) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java]:
```suggestion
    @PostMapping
```

File: src/main/java/com/dev/cinema/controller/CinemaHallController.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public void create(@RequestBody CinemaHallRequestDto requestDto) {
+        cinemaHallService.add(cinemaHallMapperImpl.toEntity(requestDto));
+    }
+
+    @RequestMapping()
```
Comment (boroda4436) [inline src/main/java/com/dev/cinema/controller/CinemaHallController.java]:
```suggestion
    @RequestMapping
```

File: src/main/java/com/dev/cinema/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @PostMapping
+    public void create(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.add(sessionMapper.toEntity(movieSessionRequestDto));
+    }
+
+    @RequestMapping("/available")
+    public List<MovieSessionResponseDto> getAvailableSession(@RequestParam(value =
+            "movieId") Long id, @RequestParam(value = "date") String showTime) {
```
Comment (boroda4436) [inline src/main/java/com/dev/cinema/controller/MovieSessionController.java]:
```suggestion
            "movieId") Long id, @RequestParam(value = "date") LocalDateTime showTime) {
```

**Review by boroda4436 — CHANGES_REQUESTED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**